tests/misc/ContextManagersTest.py

```diff
@@ -108,6 +108,15 @@ def test_make_temp(self):
             self.assertTrue(f_b.endswith(".orig"))
             self.assertTrue(os.path.basename(f_b).startswith("pre"))
 
+    def test_temp_file_existence(self):
+        """
+        Test that the temporary file created exists only within the with
+        statement context and not outside it
+        """
+        with make_temp() as temporary:
+            self.assertTrue(os.path.isfile(temporary))
+        self.assertFalse(os.path.isfile(temporary))
```

Comment:
You should move `self.assertFalse(...)` to `test_make_temp` above, below the first `with`-statement, then you can delete the rest :)